### DISCUSSION TITLE
React app with React Router deployed to Github Pages Unable to access a particular Route

### DISCUSSION THREAD

**Discussion by DexterCodes5** [community/community#55735] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

In the App.jsx I have the following routes:
`<BrowserRouter basename={import.meta.env.DEV ? '/' : '/fixerpro-frontend/'}>
      <Routes>
        <Route path="/" element={<Layout cart={cart} />}>
          <Route path="/" element={<Home vehicle={vehicle} setVehicle={setVehicle} />} />
          <Route path="cart" element={<CartPage cart={cart} setCart={setCart} user={user} checkoutState={checkoutState} />} />
        </Route>
        <Route path="login" element={<LoginPage user={user} setUser={setUser} setEmailVerificationPath={setEmailVerificationPath}/>} />
   </Routes>
</BrowserRouter>`
I can't access the login route, the other routes inside the route with path="/" work. When running the app with `npm run dev`  on localhost the login route works. How can I make this route work? The path should be: https://dextercodes5.github.io/fixerpro-frontend/login. The configuration seems ok.

**Comment by anetrda** [comment]
https://www.freecodecamp.org/news/deploy-a-react-app-to-github-pages/
i don't know much about react router or git, but in this tutorial they say you need hash router instead of browser router, more info in that tutorial
hope it helps

**Reply by DexterCodes5** [reply]
I deployed to Netlify, still don't know how to deploy to Github Pages with Vite and React. Hash Router is deprecated and if you need to change your code to push it into production something is not right.

**Comment by mccormickk** [comment]
To make the login route work properly with the specified path
Modify the basename prop in your BrowserRouter component to reflect the correct path:
<BrowserRouter basename={import.meta.env.DEV ? '/' : '/fixerpro-frontend'}>

Update the path prop for the login route to include the correct relative path:
<Route path="/fixerpro-frontend/login" element={<LoginPage user={user} setUser={setUser} setEmailVerificationPath={setEmailVerificationPath}/>} />

I'm no good at React, but I hope this helps you

**Reply by DexterCodes5** [reply]
Yes but I had multiple routes and I used anchor tags <a> to redirect between them. This may work for SPA, but in my case it didn't. I have the solution, you convert Vite React App to Create React App and you upload to Netlify by following this tutorial: https://www.freecodecamp.org/news/how-to-deploy-react-router-based-app-to-netlify/
How to deploy a Vite React App to Github Pages, you deploy a Create React App to Netlify.
